docs/4.1. Behaviour - Authorization Servers.md

```diff
@@ -36,14 +36,20 @@ client type and redirect URIs to be provided.
 
 Clients and Authorization Servers MUST provide for the OAuth 2.0 Dynamic Client Registration Protocol
 [RFC 7591][RFC-7591] to allow dynamic registration of clients to the Authorization Server to allow for
-zero-configuration setup. This endpoint MUST be protected by a transport-layer security mechanism.
+zero-configuration setup.
 
 Authorization Servers MAY additionally support the OAuth 2.0 Dynamic Client Registration Management Protocol
 [RFC 7592][RFC-7592] to allow clients to check and modify existing registrations.
 
-Whilst it is RECOMMENDED that clients provide a means of authorization when registering dynamically (which is beyond
-the scope of this specification), Section 3 of [RFC 7591][RFC-7591] permits that clients may register with no
-authorization. The client registration endpoint SHOULD be rate-limited to prevent Denial-of-Service attacks.
+Authorization Servers MUST support authorization of dynamic client registrations via manual operator approval and MAY
+support additional mechanisms including:
+*   Presentation of a valid Client TLS certificate
+*   An initial JSON Web Token (JWT) as suggested in Section 3 of [RFC 7591][RFC-7591]
```

Comment:
> Alternatively, if the server supports authorized registration, the
   developer or the client will be provisioned with an initial access
   token.  (The method by which the initial access token is obtained is
   out of scope for this specification.)

Have anyone ever tested this? How can a node tell if this is required?

Reply:
A user generates an initial access token and installs it on the Node via some vendor-specific mechanism. The Node presents it to the Auth Server. So both Auth Server and Node would have to support this, which I don't think we should mandate. It's just another option. Keycloak seems to support it - https://www.keycloak.org/docs/latest/securing_apps/#_initial_access_token. Something to try out!

Reply:
There's another RFC very much related to his, I mentioned [here](https://github.com/AMWA-TV/nmos-authorization/issues/62#issuecomment-693609891)

Reply:
The use of an initial access token is interesting, but perhaps not in the way Keycloak requires it at present. The dynamic client registration RFC talks about a 'software statement' JWT which I had somewhat dismissed previously due to bootstrapping concerns, but having looked again this has parallels with the client certificate mechanism we're using in the EST spec. A token built into the software and signed by the vendor could be used as the initial authorization mechanism, where the Authorization Server simply verifies the signature.

Reply:
Discussed on call.  Further checks on Keycloak mechanism and how well it's supported needed or alternatively more investigation of initial access token.

Reply:
Discussed on call. Support for software statement on OpenID's list of certifiers not currently good (though info is old) -- could be preferred method with fallback methods allowed.   To be continued...